contracts/prover/BaseProver.sol

```diff
@@ -2,28 +2,40 @@
 pragma solidity ^0.8.13;
 
 import {IProver} from "../interfaces/IProver.sol";
+import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
 
 /**
  * @title BaseProver
  * @notice Base implementation for intent proving contracts
  * @dev Provides core storage and functionality for tracking proven intents
  * and their claimants
  */
-abstract contract BaseProver is IProver {
+abstract contract BaseProver is IProver, ERC165 {
+    address public immutable INBOX;
     /**
      * @notice Mapping from intent hash to address eligible to claim rewards
      * @dev Zero address indicates intent hasn't been proven
      */
     mapping(bytes32 => address) public provenIntents;
 
-    /**
-     * @notice Gets the address eligible to claim rewards for a given intent
-     * @param intentHash Hash of the intent to query
-     * @return Address of the claimant, or zero address if unproven
-     */
-    function getIntentClaimant(
-        bytes32 intentHash
-    ) external view override returns (address) {
-        return provenIntents[intentHash];
+    constructor(address _inbox) {
+        // Initialize the inbox address
+        INBOX = _inbox;
+    }
+
+    function destinationProve(
+        address _sender,
+        uint256 _sourceChainId,
```

Comment:
The supportsInterface implementation could benefit from caching to improve gas efficiency, especially since this will be called frequently for verification.